File: app/src/main/java/com/tbm/dogs/activities/ViecDangCho.java
package com.tbm.dogs.activities;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.ExpandableListView;

import com.tbm.dogs.R;
import com.tbm.dogs.adapters.AdapterCongViec;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class ViecDangCho extends AppCompatActivity {
    private static final String TAG = "ViecDangcho";
    private ExpandableListView expandableListView;
    private HashMap<String,List<String>> data;

    private void init(){
        expandableListView = findViewById(R.id.expan_cong_viec);
    }
    private void initList(){
        //prepare data

        //data for header group
        final List<String> listHeader = new ArrayList<>();
        listHeader.add("Giao hàng nhanh");
        listHeader.add("Cứu hộ");
        listHeader.add("Vận chuyển đồ");



        //data for child
        data = new HashMap<>();
        List<String> listGiaoHang = new ArrayList<>();
        listGiaoHang.add("chuyển đồ tới ictu");
        listGiaoHang.add("ship hàng tới xyz");

        List<String> listCuuHo = new ArrayList<>();
        listCuuHo.add("Hết xăng");
        listCuuHo.add("Hỏng xe");

        List<String> listVanChuyenDo = new ArrayList<>();
        listVanChuyenDo.add("Chuyen phongjdafkjaskdfjasljgjdafkjaskdfjasljkfdljalsdjflkajgjdafkjaskdfjasljkfdljalsdjflkajgjdafkjaskdfjasljkfdljalsdjflkajgjdafkjaskdfjasljkfdljalsdjflkajkfdljalsdjflkajsldkfjaslkjdfljalfjlkasjdklfsdf");
        listVanChuyenDo.add("Chuyen nha tro");




        data.put(listHeader.get(0), listGiaoHang);
        data.put(listHeader.get(1), listCuuHo);
        data.put(listHeader.get(2), listVanChuyenDo);

        //setup adapter for ExpandableListView
        AdapterCongViec adapter = new AdapterCongViec(this, listHeader, data);
        expandableListView.setAdapter(adapter);
        expandableListView.setOnChildClickListener(new ExpandableListView.OnChildClickListener() {
            @Override
            public boolean onChildClick(ExpandableListView parent, View v, int groupPosition, int childPosition, long id) {
                Log.e(TAG, "onChildClick: " + listHeader.get(groupPosition) + ", " + data.get(listHeader.get(groupPosition)).get(childPosition));
                return true;
            }
        });

        expandableListView.setOnGroupClickListener(new ExpandableListView.OnGroupClickListener() {
            @Override
            public boolean onGroupClick(ExpandableListView parent, View v, int groupPosition, long id) {
                Log.e(TAG, "onGroupClick: " + groupPosition);
                return false;
            }
        });

    }
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getSupportActionBar().hide();
        setContentView(R.layout.activity_viec_dang_cho);
        init();
        initList();
    }
}

File: app/src/main/java/com/tbm/dogs/activities/ViecDangCho.kt
package com.tbm.dogs.activities

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.widget.ExpandableListView
import com.tbm.dogs.Helper.Var
import com.tbm.dogs.R
import com.tbm.dogs.adapters.AdapterCongViec
import com.tbm.dogs.model.obj.Job
import java.util.*

class ViecDangCho : AppCompatActivity() {
    lateinit var expandableListView: ExpandableListView
    lateinit var data: HashMap<String, List<Job>>

    private fun init() {
        expandableListView = findViewById(R.id.expan_cong_viec)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
    }

    private fun initList() {
        //data for header group
        val listHeader = ArrayList<String>()
        listHeader.add("Giao hàng nhanh")
//        listHeader.add("Cứu hộ")
//        listHeader.add("Vận chuyển đồ")


        //data for child
        data = HashMap()
        val listGiaoHang = ArrayList<Job>()
        if(Var.jobsWaiting != null){
            listGiaoHang.addAll(Var.jobsWaiting!!)
        }

//
//        val listCuuHo = ArrayList<String>()
//        listCuuHo.add("Hết xăng")
//        listCuuHo.add("Hỏng xe")
//
//        val listVanChuyenDo = ArrayList<String>()
//        listVanChuyenDo.add("Chuyen phongjdafkjaskdfjasljgjdafkjaskdfjasljkfdljalsdjflkajgjdafkjaskdfjasljkfdljalsdjflkajgjdafkjaskdfjasljkfdljalsdjflkajgjdafkjaskdfjasljkfdljalsdjflkajkfdljalsdjflkajsldkfjaslkjdfljalfjlkasjdklfsdf")
//        listVanChuyenDo.add("Chuyen nha tro")




        data[listHeader[0]] = listGiaoHang
//        data[listHeader[1]] = listCuuHo
//        data[listHeader[2]] = listVanChuyenDo

        //setup adapter for ExpandableListView
        val adapter = AdapterCongViec(this, listHeader, data)
        expandableListView.setAdapter(adapter)
        expandableListView.setOnChildClickListener { parent, v, groupPosition, childPosition, id ->
            Log.e(TAG, "onChildClick: " + listHeader[groupPosition] + ", " + data[listHeader[groupPosition]]!!.get(childPosition))
            true
        }

        expandableListView.setOnGroupClickListener { parent, v, groupPosition, id ->
            Log.e(TAG, "onGroupClick: $groupPosition")
            false
        }

    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_viec_dang_cho)
        init()
        initList()
    }

    companion object {
        private val TAG = "JobWaiting"
    }
}
